### PR TITLE
HackStudio: Add ability to build and run serenity components

### PR DESCRIPTION
**Pull request by itamar8910**
When opening the serenity repository in Hack Studio, you can now build and run standalone components.
Steps:
1. Install the `cmake` and `gcc` ports.
2. Run `Meta/serenity.sh copy-src` to install the serenity repository into `~/Source/serenity` in the disk image.
3. Open `~/Source/serenity` with Hack Studio.
4. Go to a source file belonging to a serenity component.
5. Build and run with the existing Hack Studio UI (Building won't re-build the dependencies, just link against them).

This enables editing the serenity source from Hack Studio, building in-system and seeing the results. Pretty cool :^)

Note that currently not all components succeed to build. A common issue I encountered is failure to build things that use `LibConfig` because it uses generated header files which are not currently generated.

An example with building `Breakout`:
hs-breakout2

### CODE REVIEW

File: Userland/Libraries/LibCore/TempFile.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    for (int i = filename.size() - 1; i >= 0; --i) {
+        if (filename[i] != 'X')
+            break;
+        filename[i] = random_characters[(get_random<u32>() % (sizeof(random_characters) - 1))];
+    }
+
+    return String(filename.data(), filename.size());
+}
+
+TempFile::TempFile(Type type)
```
Comment (awesomekling) [inline Userland/Libraries/LibCore/TempFile.cpp:38]: Can we implement this on top of `mkstemp()` instead?

File: Userland/Libraries/LibCore/Command.h [hunk 2]
```diff
@@ -14,7 +14,14 @@
...
-String command(const String& command_string, Optional<LexicalPath> chdir);
+
+struct CommandResult {
+    int exit_code { 0 };
+    String stdout;
+    String stderr;
+};
+
+Optional<CommandResult> command(const String& program, const Vector<String>& arguments, Optional<LexicalPath> chdir);
+Optional<CommandResult> command(const String& command_string, Optional<LexicalPath> chdir);
```
Comment (awesomekling) [inline Userland/Libraries/LibCore/Command.h:25]: Would this be nicer as `ErrorOr<CommandResult>`?
Also, east const style pls. :^)

File: Userland/DevTools/HackStudio/Project.cpp [hunk 3]
```diff
@@ -59,4 +59,9 @@ String Project::to_absolute_path(String const& path) const
     return LexicalPath { String::formatted("{}/{}", m_root_path, path) }.string();
 }
 
+bool Project::project_is_serenity() const
+{
+    return m_root_path.ends_with("Source/serenity");
```
Comment (awesomekling) [inline Userland/DevTools/HackStudio/Project.cpp:64]: This heuristic is a little bit weird, but all right :^)

**Comment by awesomekling** [comment]
This is super cool!

File: Userland/DevTools/HackStudio/Project.cpp (hunk 3 above)
Comment (itamar8910) [reply Userland/DevTools/HackStudio/Project.cpp:66]: It is yeah. Added a comment & a FIXME for it.

**Pull request merged**